File: Doubly_linked_list/DoublyLinkedListCountNodes.java
// File Name: DoublyLinkedListCountNodes.java
// Program to count the number of nodes in a doubly linked list

class Node {
    int data;
    Node prev;
    Node next;

    Node(int data) {
        this.data = data;
        this.prev = null;
        this.next = null;
    }
}

public class DoublyLinkedListCountNodes {
    static Node head = null;

    // Function to count nodes
    static int count_nodes() {
        int count = 0;
        Node temp = head;

        while (temp != null) {
            count++;
            temp = temp.next;
        }
        return count;
    }

    public static void main(String[] args) {
        System.out.println("Number of nodes = " + count_nodes());
    }
}

File: Doubly_linked_list/doubly_linked_list_count_nodes.py
# File Name: doubly_linked_list_count_nodes.py
# Program to count the number of nodes in a doubly linked list

class Node:
    def __init__(self, data):
        self.data = data
        self.prev = None
        self.next = None

head = None

# Function to count nodes
def count_nodes():
    global head
    count = 0
    temp = head

    while temp is not None:
        count += 1
        temp = temp.next

    return count

if __name__ == "__main__":
    print("Number of nodes =", count_nodes())
